### PR TITLE
COMMERCE-11494

### PR DESCRIPTION
**Pull request by Brunno0liveira**
[COMMERCE-11494](https://liferay.atlassian.net/browse/COMMERCE-11494) [COMMERCE-11491](https://liferay.atlassian.net/browse/COMMERCE-11491)

### CODE REVIEW

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 1]
```diff
@@ -1013,6 +1013,143 @@ definition {
...
+		property portal.acceptance = "false";
+
+		task ("Given a Minium site is created") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("And a simple product with a new option is created") {
+			CommerceNavigator.gotoPortlet(
+				category = "Product Management",
+				portlet = "Products");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1029]: Use direct URL navigation instead

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 2]
```diff
@@ -1013,6 +1013,143 @@ definition {
...
+
+			CPCommerceCatalog.newProductsEntry(
+				productCatalog = "Minium",
+				productName = "Simple T-Shirt",
+				productType = "Simple");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1034]: Can you do this using APIs?

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 3]
```diff
@@ -1013,6 +1013,143 @@ definition {
...
+
+			SelectFrameTop();
+
+			Click(locator1 = "Icon#CLOSE");
+		}
+
+		task ("And from the Product Management -> Options, the option is deleted") {
+			CommerceNavigator.gotoPortlet(
+				category = "Product Management",
+				portlet = "Options");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1057]: Use direct URL navigation instead

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 4]
```diff
@@ -1013,6 +1013,143 @@ definition {
...
+			CommerceProducts.openProductsAdmin();
+
+			CommerceNavigator.searchEntry(entryName = "Simple T-Shirt");
+
+			CommerceNavigator.gotoEntry(entryName = "Simple T-Shirt");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Options");
+		}
+
+		task ("Then the admin can create a new product option") {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1076]: Assert here that the previous option was deleted

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 5]
```diff
@@ -1013,6 +1013,143 @@ definition {
...
+			CommerceProductOptions.editProductOption(
+				key_fieldType = "Select from List",
+				key_option = "New Option",
+				productOptionFieldType = "true",
+				productOptionRequired = "true",
+				productOptionSkuContributor = "true");
+
+			SelectFrameTop();
+
+			Click(locator1 = "Icon#CLOSE");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1051]: Is this actually needed for the test?

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 6]
```diff
@@ -1013,6 +1013,143 @@ definition {
...
+		}
+
+		task ("Then the admin can create a new product option") {
+			CommerceProductOptions.createProductOption(key_optionName = "New Option1");
+		}
+	}
+
+	@description = "COMMERCE-11494. Can delete product option with generate All SKUs are created"
+	@priority = 4
+	test CanDeleteProductOptionWithGenerateAllSKUsCreated {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1083]:
```suggestion
	test CanDeleteProductOptionWithGeneratedSkus {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 7]
```diff
@@ -1013,6 +1013,143 @@ definition {
...
+
+			CommerceEntry.gotoMenuTab(menuTab = "Options");
+		}
+
+		task ("Then the admin can create a new product option") {
+			CommerceProductOptions.createProductOption(key_optionName = "New Option1");
+		}
+	}
+
+	@description = "COMMERCE-11494. Can delete product option with generate All SKUs are created"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1081]:
```suggestion
	@description = "COMMERCE-11494. Can delete product option with all skus generated"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 8]
```diff
@@ -1013,6 +1013,143 @@ definition {
...
+	@priority = 4
+	test CanDeleteProductOptionWithGenerateAllSKUsCreated {
+		property custom.properties = "jsonws.web.service.paths.excludes=${line.separator}company.security.strangers.verify=false";
+		property portal.acceptance = "false";
+
+		task ("Given a Minium site is created") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("And an option is created") {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1091]:
```suggestion
		task ("And a product option is created") {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 9]
```diff
@@ -1013,6 +1013,143 @@ definition {
...
+			CommerceProducts.openProductsAdmin();
+
+			CommerceProducts.searchAndGoToProductEntry(entryProductName = "ABS Sensor");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Options");
+
+			CommerceProductOptions.createProductOption(key_optionName = "Option1");
+
+			Refresh();
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1100]: Please avoid using the Refresh

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 10]
```diff
@@ -1013,6 +1013,143 @@ definition {
...
+				key_value = "Option1",
+				locator1 = "CommerceEntry#BODY_VERTICAL_ELLIPSIS");
+
+			CommerceEntry.clickNoError(menuItem = "Delete");
+		}
+
+		task ("Then when the admin goes back to the products admin page and the product is displayed properly") {
+			CommerceProducts.openProductsAdmin();
+
+			CommerceNavigator.searchEntry(entryName = "ABS Sensor");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1149]: Add a couple more assertions here, for example: check a couple products more, the number of products

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]

**Comment by lorenzolapescara** [comment]
@Brunno0liveira please sort the commits so that the development commits are before the functional test commit, thanks!

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:relevant

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase (hunk 10 above)
Comment (lorenzolapescara) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1142]: @Brunno0liveira please add assertions here, do not use `CommerceNavigator.searchEntry()` to assert.

Also, by looking at the bug's description, it says:
> Expected Result:  Disable the active SKU & All the product pages shows correctly all the products

You also need to check that the skus are disabled.

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:relevant

**Comment by Brunno0liveira** [comment]
ci:test:relevant

**Review by lorenzolapescara — COMMENTED** [review]
Discussed a change via Slack, @Brunno0liveira will update the PR

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:relevant

**Comment by Brunno0liveira** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 11]
```diff
@@ -1013,6 +1013,156 @@ definition {
...
+				key_entryTitle = "VALUE1",
+				key_label = "Inactive",
+				locator1 = "CommerceEntry#SKU_STATUS_PRODUCT_ADMIN_PAGE");
+
+			CommerceProducts.searchProduct(search = "Value2");
+
+			AssertElementPresent(
+				key_entryTitle = "VALUE2",
+				key_label = "Inactive",
+				locator1 = "CommerceEntry#SKU_STATUS_PRODUCT_ADMIN_PAGE");
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:1162]: Please use a FOR cycle to do the checks.
The `key_entryTitle` value inside the assert and the value inside the search are the same. The only difference is that inside the macro is in uppercase.
For the cycle, you can use the `StringUtil.upperCase` to transform the variable in all uppercase characters

Take a look inside the code to understand how it is working.

**Review by pietrobocale — CHANGES_REQUESTED** [review]
Hey @Brunno0liveira, I asked for some changes. Please take a look at the comments.

Thanks!

**Comment by Brunno0liveira** [comment]
ci:test:relevant

**Review by pietrobocale — APPROVED** [review]
Review approved. Please check the CI to ensure that the tests are passing.

Thanks!

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Pull request closed without merging**